### ISSUE TITLE
shorthand_operator false negative if line contains non-semantic spaces after certain statements

### ISSUE THREAD

**Issue by robinkunde** [realm/SwiftLint#3071]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

If a line that would normally trigger the `shorthand_operator` rule contains any non-semantic spaces after the statement with a precedence (`*` or `/`), the rule will not trigger. For example, if you try to disable the rule with an inline comment, you will instead get a `superflous_disable_command` violation.

Examples:
```
lhs = lhs / rhs // even if the line only has a trailing space, the rule does not trigger
lhs = lhs / rhs // swiftlint:disable:this shorthand_operator
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ echo "var test = 1\n\ntest = test / 1" | swiftlint lint --no-cache --use-stdin
Loading configuration from '.swiftlint.yml'
<nopath>:3:1: error: Shorthand Operator Violation: Prefer shorthand operators (+=, -=, *=, /=) over doing the operation and assigning. (shorthand_operator)
Done linting! Found 1 violation, 1 serious in 1 file.

$ echo "var test = 1\n\ntest = test / 1 //" | swiftlint lint --no-cache --use-stdin
Loading configuration from '.swiftlint.yml'
Done linting! Found 0 violations, 0 serious in 1 file.

$ echo "var test = 1\n\ntest = test / 1 // swiftlint:disable:this shorthand_operator" | swiftlint lint --no-cache --use-stdin
Loading configuration from '.swiftlint.yml'
<nopath>:3:1: warning: Superfluous Disable Command Violation: SwiftLint rule 'shorthand_operator' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.38.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
tried Homebrew, Cocoapods, and master
* Paste your configuration file:
default config, see above

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
no
* Which Xcode version are you using (check `xcodebuild -version`)?
11.3.1
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
var test = 1

test = test / 1 // this should trigger but doesn't
```
